core/chain_util.go

```diff
@@ -346,3 +349,34 @@ func GetBlockByHashOld(db ethdb.Database, hash common.Hash) *types.Block {
 	}
 	return (*types.Block)(&block)
 }
+
+// returns a formatted MIP mapped key.
+func mipmapKey(num, level uint64) []byte {
+	key := new(big.Int).SetUint64(num / level * level)
+
+	return append(mipmapPre, key.Bytes()...)
+}
```

Comment:
Ok, but then the #0 bloom filter will contain all the receipts up till 1M, irrelevant of the level requested.